pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -360,40 +361,67 @@ private ParsedTimestamp parseBackendTimestamp(String str) throws SQLException {
 
     } catch (NumberFormatException nfe) {
       throw new PSQLException(
-          GT.tr("Bad value for type timestamp/date/time: {0}", str),
+          GT.tr("Bad value for type timestamp/date/time: {0}", new String(s)),
           PSQLState.BAD_DATETIME_FORMAT, nfe);
     }
 
     return result;
   }
 
+  ParsedTimestamp parseDate(byte[]dateBytes) {
+    ParsedTimestamp parsedTimestamp = new ParsedTimestamp();
+    int length = dateBytes.length;
+
+    if (dateBytes[length - 2] == 'B' && dateBytes[length - 1] == 'C') {
+      length = length - 3;
+      parsedTimestamp.era = GregorianCalendar.BC;
+    }
+    int var1 = 0;
+    for (parsedTimestamp.year = 0; dateBytes[var1] != 45; parsedTimestamp.year = parsedTimestamp.year * 10 + (dateBytes[var1++] - 48)) {
+    }
+
+    ++var1;
+
+    for (parsedTimestamp.month = 0; dateBytes[var1] != 45; parsedTimestamp.month = parsedTimestamp.month * 10 + (dateBytes[var1++] - 48)) {
+    }
+
+    ++var1;
+
+    for (parsedTimestamp.day = 0; var1 < length; parsedTimestamp.day = parsedTimestamp.day * 10 + (dateBytes[var1++] - 48)) {
+
+    }
+
+    return parsedTimestamp;
+  }
+
   /**
    * Parse a string and return a timestamp representing its value.
    *
    * @param cal calendar to be used to parse the input string
-   * @param s The ISO formated date string to parse.
-   * @return null if s is null or a timestamp of the parsed string s.
+   * @param bytes The ISO formated date string to parse.
+   * @return null if bytes is null or a timestamp of the parsed bytes bytes.
    * @throws SQLException if there is a problem parsing s.
    */
   public @PolyNull Timestamp toTimestamp(@Nullable Calendar cal,
-      @PolyNull String s) throws SQLException {
+      byte @PolyNull []bytes) throws SQLException {
+
     try (ResourceLock ignore = lock.obtain()) {
-      if (s == null) {
+      if (bytes == null) {
         return null;
       }
 
-      int slen = s.length();
+      int blen = bytes.length;
 
       // convert postgres's infinity values to internal infinity magic value
-      if (slen == 8 && "infinity".equals(s)) {
+      if (bytes[0] == 'i' && Arrays.equals(bytes, "infinity".getBytes())) {
```

Comment:
`"infinity".getBytes()` relies on the default encoding, and it would allocate a new byte array on every execution. I suggest saving `"infinity".getBytes()` into a `private static final byte[]` array. The same goes for `-infinity`